crates/solver/src/settlement_submission/flashbots_api.rs

```diff
@@ -26,6 +26,39 @@ struct FlashbotGasPrice {
     default: Eip1559,
 }
 
+#[derive(Debug, Deserialize)]
+struct FlashbotStatus {
+    status: Status,
+}
+
+#[derive(Debug, Deserialize, PartialEq)]
+enum Status {
+    #[serde(rename = "PENDING_BUNDLE")]
+    Pending,
+    #[serde(rename = "FAILED_BUNDLE")]
+    Failed,
+    #[serde(rename = "SUCCESSFUL_BUNDLE")]
+    Successful,
+    #[serde(rename = "CANCEL_BUNDLE_SUCCESSFUL")]
+    Cancelled,
+}
+
+fn parse_json_rpc_response<T>(body: &str) -> Result<T>
+where
+    T: DeserializeOwned,
+{
+    serde_json::from_str::<Output>(body)
+        .with_context(|| {
+            tracing::info!("flashbot response: {}", body);
+            anyhow!("invalid flashbots response")
+        })
+        .and_then(|output| match output {
+            Output::Success(body) => serde_json::from_value::<T>(body.result)
+                .context("flashbots failed conversion to expected type"),
```

Comment:
if you want to be fancy you can:
```suggestion
                .with_context(|_| format!("flashbots failed conversion to expected {}", std::any::type_name::<T>())),
```
Although, I'm 100% OK with how it is now (the expected type should in theory be captured by virtue of the stack trace which `anyhow::Error`s create).

Reply:
I wanna be fancy :) Added.